tutorials/performance/pipeline_compilations.rst

```diff
@@ -0,0 +1,77 @@
+Reducing Stutter from Shader (Pipeline) Compilations
+====================================================
+
+Pipeline compilation, also commonly known as shader compilation, is an expensive operation required by the engine to be able to draw any kind of content with the GPU. 
+
+In more precise terms, *Shader Compilation* involves the translation of the GLSL code that Godot generates into an intermediate format that can be shared across systems (such as SPIR-V when using Vulkan). However, this format can't be used by the GPU directly. *Pipeline Compilation* is the step the GPU driver performs where it'll convert the intermediate shader format (the result from shader compilation) to something the GPU can actually use for rendering. Drivers usually keep a cache of pipelines stored somewhere in the system to avoid repeating the process every time a game is run. This cache is usually deleted when the driver is updated.
```

Comment:
This section might be made more clear by explicitly listing out the shader compilation stages, ideally with some graphic:
`VisualShader`/`StandardMaterial` (this generates GDShader in some cases, right?) -> `GDShader` (Godot Shading Language) -> `GLSL` -> intermediate format (`SPIR-V`) -> pipeline. 

As is, this assumes some knowledge of the shader compilation process (the fact that Godot compiles to GLSL as an intermediate step isn't commonly referenced throughout the rest of the docs, IIRC).

Reply:
Something like this
![ShaderCompilationSteps](https://github.com/user-attachments/assets/9bab6448-d5f6-493b-ad79-c5c2058da95c)

(I'm not sure if all this is accurate, and labelling just the single step as "shader compilation" may be misleading")

Reply:
I like it! Can I include this graph as is? Or do we need to handle it in some way to take transparency or background into account?

Reply:
Delaying for later.